pandas/core/groupby/groupby.py

```diff
@@ -1171,10 +1171,18 @@ def std(self, ddof=1, *args, **kwargs):
         ddof : integer, default 1
             degrees of freedom
         """
-
-        # TODO: implement at Cython level?
         nv.validate_groupby_func('std', args, kwargs)
-        return np.sqrt(self.var(ddof=ddof, **kwargs))
+        if ddof == 1:
+            try:
+                return self._cython_agg_general('std', **kwargs)
+            except Exception:
```

Comment:
Not entirely clear what this is for but except for legacy code we never catch just base Exception so will need something more explicit

Reply:
I copy-pasted the implementation of the method `groupby.var`, replacing `var` with `std`.

Reply:
please compare ::
https://github.com/xieyuheng/pandas/blob/master/pandas/core/groupby/groupby.py#L1163
https://github.com/xieyuheng/pandas/blob/master/pandas/core/groupby/groupby.py#L1189

Reply:
My PR solved a three years old bug, please merge it.

Reply:
Copy / paste isn't the right way to go here. Can you still keep the dispatch to `self.var` and just see what is mangling the expression once passed as an argument to `np.sqrt` instead?